.github/workflows/ci.yml

```diff
@@ -30,105 +30,40 @@ jobs:
-            { platform: android, arch: arm32, os: ubuntu-22.04 },
-            { platform: android, arch: x86_64, os: ubuntu-22.04 },
-            { platform: android, arch: x86_32, os: ubuntu-22.04 },
-            { platform: ios, arch: arm64, os: macos-latest },
-            { platform: web, arch: wasm32, os: ubuntu-22.04 }
-          ]
-        target-type: [editor, template_debug, template_release]
-        # disabling double precision builds for now; doubles the CI time for
-        # unlikely case.
-        float-precision: [single] # , double]
+    name: Build GDExtension
+    uses: ./.github/workflows/build.yml
 
-    runs-on: ${{ matrix.target.os }}
+  test:
+    name: Run tests
+    needs: build
+    runs-on: ubuntu-22.04
+    env:
+      ARTIFACT_NAME: ${{ needs.build.outputs.artifact-name }}
+      ADDON_PATH: demo/addons/${{ needs.build.outputs.artifact-name }}
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: true
-
-      # Lint
-      #- name: Setup clang-format
-      #  shell: bash
-      #  run: |
-      #    python -m pip install clang-format
-      #- name: Run clang-format
-      #  shell: bash
-      #  run: |
-      #    clang-format src/** --dry-run --Werror
+      - name: Checkout
+        uses: actions/checkout@v4
 
-      # Setup dependencies
-      - name: Setup godot-cpp build environment
-        uses: godotengine/godot-cpp/.github/actions/setup-godot-cpp@godot-4.4-stable
+      - name: Download Binaries
+        uses: actions/download-artifact@v4
         with:
-          platform: ${{ matrix.target.platform }}
-          em-version: 3.1.62
-
-      # Build GDExtension (with caches)
-      - name: Cache .scons_cache
-        uses: actions/cache@v4
-        with:
-          path: ${{ github.workspace }}/.scons-cache/
-          key: scons_cache-${{ matrix.target.platform }}_${{ matrix.target.arch }}_${{ matrix.float-precision }}_${{ matrix.target-type }}_${{ hashFiles('.gitmodules') }}
-
-      - name: Build GDExtension
-        shell: sh
-        env:
-          SCONS_CACHE: ${{ github.workspace }}/.scons-cache/
-        run: |
-          scons \
-            verbose=1 \
-            target_path=bin/ \
-            target=${{ matrix.target-type }} \
-            platform=${{ matrix.target.platform }} \
-            arch=${{ matrix.target.arch }} \
-            precision=${{ matrix.float-precision }}
-
-      # TODO: run gdscript tests also, see
-      # https://github.com/godotengine/godot-cpp/blob/f08e781b0ba3e45264d7d40e5d6debaa0d838383/.github/workflows/ci.yml#L142-L178
-      - name: Run doctests
-        if: matrix.target.run_tests
-        run: |
-          scons \
-            verbose=1 \
-            target=${{ matrix.target-type }} \
-            platform=${{ matrix.target.platform }} \
-            arch=${{ matrix.target.arch }} \
-            precision=${{ matrix.float-precision }} \
-            run_tests
+          pattern: ${{ env.ARTIFACT_NAME }}-*
```

Comment:
If you're not merging the build artifacts in the build task, we should just pull the ubuntu editor dll we'll actually be testing.